cvrf2csaf/cvrf2csaf.py

```diff
@@ -130,6 +132,27 @@ def convert_file(self, path) -> dict:
         return self._compose_final_csaf()
 
 
+    def validate_output_against_schema(self, output_file) -> bool:
+        """
+        Validates the CSAF output against the CSAF JSON schema
+        return: True if valid, False if invalid
+        """
+        with open(self.CSAF_SCHEMA_FILE) as f:
+            csaf_schema_content = json.loads(f.read())
+
+        with open(output_file) as f:
+            output_file_content = json.loads(f.read())
+
+        try:
+            validate(output_file_content, csaf_schema_content)
```

Comment:
This should include the Format checker for Draft 2020-12

Reply:
Can you elaborate? I am using this schema for the validation:

https://docs.oasis-open.org/csaf/csaf/v2.0/csaf_json_schema.json

Reply:
Maybe import explicitly `Draft202012Validator` and use that instead of the generic `validate` function?

Reply:
One of those could work:
```
validate(output_file_content, csaf_schema_content,  jsonschema.draft202012_format_checker)
```

```
schema_store[csaf_schema_content['$id']] = csaf_schema_content

resolver = RefResolver.from_schema(schema_store[list(schema_store)[0]], store=schema_store)
validator = Draft202012Validator(schema, resolver, jsonschema.draft202012_format_checker)
validator.validate(output_file_content)
```

I haven't tested the code - just from the top of my head...
You might need to load the referenced schemas as well:
```
    for i in json_referenced_schemas:
        with open(i, 'r') as f:
            current_ref_schema_data = f.read()
            current_ref_schema = json.loads(current_ref_schema_data)
            schema_store[current_ref_schema['$id']] = current_ref_schema
```

Reply:
Please check [this once again](https://github.com/csaf-tools/CVRF-CSAF-Converter/issues/45#issuecomment-1031382756). Both is needed.

Reply:
Done.

Draft2020 now being used also with format_checker.

The references schemas I found are these:
```
              "properties": {
                "cvss_v2": {
                  "$ref": "https://www.first.org/cvss/cvss-v2.0.json"
                },
                "cvss_v3": {
                  "oneOf": [
                    {
                      "$ref": "https://www.first.org/cvss/cvss-v3.0.json"
                    },
                    {
                      "$ref": "https://www.first.org/cvss/cvss-v3.1.json"
                    }
                  ]
                },
```

And these are being loaded automatically upon calling the main validate function (tested with the cvss), so no need to load them.

Reply:
Also now using strict CSAF schema

Reply:
perfect, thank you.